### PR TITLE
Fix compilation of legacy linux-sunxi-3.4 kernel under zeus

### PR DESCRIPTION
**Pull request by kedder**
This fixes few compilation issues under GCC-9 on latest Zeus branch of OpenEmbedded and meta-sunxi layer. This adds a few patches to `linux-sunxi` and `u-boot-sunxi` recipes, and also suggests to disable devicetree builds when compiling in legacy mode.

### CODE REVIEW

**Comment by kedder** [comment]
I've created this PR against `master` branch, but with the intention for them to be ported to `zeus`. If you want PR against `zeus` directly, I can rebase.

File: README.md [hunk 1]
```diff
@@ -26,9 +26,10 @@ If you want to switch back to sunxi versions for some reasons (no device tree av
 - change the file conf/machine/include/sunxi.inc to include the following block
 - edit your conf/local.conf to add the following block
 
-	PREFERRED_PROVIDER_u-boot="u-boot-sunxi"
-	PREFERRED_PROVIDER_virtual/bootloader="u-boot-sunxi"
-	PREFERRED_PROVIDER_virtual/kernel="linux-sunxi"
+		PREFERRED_PROVIDER_u-boot="u-boot-sunxi"
```
Comment (nandra) [inline README.md]: can you please put alignment as previous version? thanks

File: recipes-bsp/u-boot/u-boot-sunxi.bb [hunk 2]
```diff
@@ -22,6 +22,7 @@ SRC_URI = " \
 	git://github.com/linux-sunxi/u-boot-sunxi.git;protocol=git;branch=sunxi \
 	file://0002-gcc5-fixes.patch \
```
Comment (nandra) [inline recipes-bsp/u-boot/u-boot-sunxi.bb]: if you are touching this, can you please rename patches to start from 0001? Thanks

File: recipes-kernel/linux/linux-sunxi_3.4.bb [hunk 3]
```diff
@@ -15,8 +15,11 @@ SRC_URI += "git://github.com/linux-sunxi/linux-sunxi.git;branch=sunxi-3.4;protoc
         file://0001-compiler-gcc-integrate-the-various-compiler-gcc-345-.patch \
         file://0002-use-static-inline-in-ARM-ftrace.patch \
         file://0001-gcc5-fixes.patch \
+        file://0003-arm-put-user-asm-fix.patch \
```
Comment (nandra) [inline recipes-kernel/linux/linux-sunxi_3.4.bb]: Same here, pls rename patches. Thanks.

**Comment by nandra** [comment]
We first merge to master, then you can post PR agains zeus. Thanks.

File: README.md (hunk 1 above)
Comment (kedder) [reply README.md]: I fixed alignment to make it render properly in github. Currently this part is unformatted:

image

That said, do you still want me to restore previous indentation?

File: recipes-bsp/u-boot/u-boot-sunxi.bb (hunk 2 above)
Comment (kedder) [reply recipes-bsp/u-boot/u-boot-sunxi.bb]: Done.

File: recipes-kernel/linux/linux-sunxi_3.4.bb (hunk 3 above)
Comment (kedder) [reply recipes-kernel/linux/linux-sunxi_3.4.bb]: Done.

File: README.md (hunk 1 above)
Comment (nandra) [reply README.md]: Sorry I overlooked it. No keep it like you have it in PR.

**Comment by nandra** [comment]
If you need it in zeus pls post updated PR against zeus branch. Out of curiosity which board you're running this old kernel? Thanks.

**Comment by kedder** [comment]
> which board you're running this old kernel

I'm using Cubieboard 2 for https://openvario.org/ project. It requires mali for graphics, so I'm trying to get the old kernel running on new OE build system. 

It seems like there is a distant possibility to use newer 5.2 with OSS driver. But I need a working fresh build before upgrading the kernel.

**Comment by nandra** [comment]
If you look at here you can find some info about lima (open source mali support): https://github.com/linux-sunxi/meta-sunxi/issues/273 but also we have in current repo mali drivers which you can try to play with. Thanks.

**Pull request merged**